tool/tsh/common/tsh.go

```diff
@@ -2086,14 +2086,32 @@ func onLogout(cf *CLIConf) error {
 		proxyHost = cf.Proxy
 	}
 
+	tc, err := makeClient(cf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	var SAMLSingleLogoutURL string
+	clt, err := tc.ConnectToCluster(cf.Context)
+	if err != nil {
+		log.
+			Warnf("Failed to retrieve cluster client for user %s, SAML single logout will be skipped: %v", active.Username, err)
+	}
+
+	user, err := clt.AuthClient.GetUser(cf.Context, active.Username, false)
+	if err != nil {
+		log.
+			Warnf("Failed to retrieve user details for user %s, SAML single logout will be skipped: %v", active.Username, err)
+	}
```

Comment:
We'll fail to log out the user if the profile expires. Should we store the SLO url (if any) in the profile yaml instead, and fetch it during login/profile hydration instead?

Reply:
We had a similar conundrum when implementing this in the WebUI where we wouldn't be able to initiate SLO if the web session expired, we decided that SLO should only be initiated in an intentional logout from an active and valid session. So if it just expired, then there will be no SLO.